source/framework/core/src/TRestHits.cxx

```diff
@@ -541,109 +759,204 @@ void TRestHits::GetBoundaries(std::vector<double>& dist, double& max, double& mi
     min -= offset * minDiff + minDiff / 2.;
     nBins = std::round((max - min) / minDiff);
 }
-
+///////////////////////////////////////////////
+/// \brief It computes the gaussian sigma in the X-coordinate.
+/// It adds a hit to the right and a hit to the left, with energy = 0 +/- 70 ADC.
+/// Then it fits a gaussian to the hits and extracts the sigma. The hits are just added
+/// for fitting purposes and do not go into any further processing.
+///
 Double_t TRestHits::GetGaussSigmaX() {
     Double_t gausSigmaX = 0;
     Int_t nHits = GetNumberOfHits();
-    vector<Double_t> x(nHits), y(nHits), ex(nHits), ey(nHits);
-    if (nHits <= 3) {
+    if (nHits <= 0) {
         gausSigmaX = 0;
     } else {
-        for (int n = 0; n < GetNumberOfHits(); n++) {
-            x[n] = fX[n];
-            y[n] = fEnergy[n];
-            ex[n] = 0;
-            if (y[n] != 0) {
-                ey[n] = 10 * sqrt(y[n]);
+        Int_t nAdd = 0;
+        bool doHitCorrection = true;
+        // bool doHitCorrection = nHits <= 18; //in case we want to apply it only to the smaller events
+        if (doHitCorrection) {
+            nAdd = 2;
+        }
+        Int_t nElems = nHits + nAdd;
+        vector<Double_t> x(nElems), y(nElems), ex(nElems), ey(nElems);
+        Int_t k = nAdd / 2;
+        Double_t xMin = std::numeric_limits<double>::max();
+        Double_t xMax = std::numeric_limits<double>::lowest();
+        for (int n = 0; n < GetNumberOfHits(); k++, n++) {
+            x[k] = fX[n];
+            y[k] = fEnergy[n];
+            ex[k] = 0;
+            xMin = min(xMin, x[k]);
+            xMax = max(xMax, x[k]);
+            if (y[k] != 0) {
+                ey[k] = 10 * sqrt(y[k]);
             } else {
-                ey[n] = 0;
+                ey[k] = 0;
             }
         }
-        TGraphErrors* grX = new TGraphErrors(nHits, &x[0], &y[0], &ex[0], &ey[0]);
-        Double_t maxY = MaxElement(nHits, grX->GetY());
-        Double_t maxX = grX->GetX()[LocMax(nHits, grX->GetY())];
+        Int_t h = nHits + nAdd / 2;
+        if (doHitCorrection) {
+            x[0] = xMin - 0.5;
+            x[h] = xMax + 0.5;
+            y[0] = 0.0;
+            y[h] = 0.0;
+            ex[0] = 0.0;
+            ex[h] = 0.0;
+            ey[0] = 70.0;
+            ey[h] = 70.0;
+        }
+        TGraphErrors* grX = new TGraphErrors(nElems, &x[0], &y[0], &ex[0], &ey[0]);
```

Comment:
You should delete the newed variables.

Reply:
You are right. Done here: https://github.com/rest-for-physics/framework/pull/254/commits/d6b6d5687292be6a6b2d43f804c37f126668aa8e